File: 001-Rat Count House/RatCountHouse.cpp
#include <iostream>

using namespace std;

int minHouses(int r, int unit, int arr[], int n) {
    if (n == 0) 
        return -1;  // Return -1 if array is empty

    int requiredFood = r * unit;
    int currentFood = 0;

    for (int i = 0; i < n; i++) {
        currentFood += arr[i];
        if (currentFood >= requiredFood){
            return i + 1; // Houses are 1-based index
        }
            
    }

    return 0; // If food is insufficient
}

int main() {
    int r = 7, unit = 2;
    int arr[] = {2, 8, 3, 5, 7, 4, 1, 2};
    int n = sizeof(arr) / sizeof(arr[0]);

    cout << minHouses(r, unit, arr, n) << endl;
    return 0;
}

File: 001-Rat Count House/RatCountHouse.py
def min_houses(r, unit, arr):
    if len(arr) == 0:
        return -1  # Return -1 if the array is empty

    required_food = r * unit
    current_food = 0

    for i in range(len(arr)):
        current_food += arr[i]
        if current_food >= required_food:
            return i + 1  # Houses are counted from 1

    return 0  # If food is insufficient

# Example usage:
r = 7
unit = 2
arr = [2, 8, 3, 5, 7, 4, 1, 2]
print(min_houses(r, unit, arr))
